arangod/Agency/State.cpp

```diff
@@ -295,32 +296,38 @@ index_t State::logNonBlocking(index_t idx, velocypack::Slice const& slice,
     }
   }
 
+  logEmplaceBackNoLock(log_t(idx, term, buf, clientId), leading);
+
+  return _log.back().index;
+}
+
+
+void State::logEmplaceBackNoLock(log_t const& l, bool leading) {
+  
   try {
-    _log.push_back(log_t(idx, term, buf, clientId));  // log to RAM or die
+    _log.emplace_back(l);  // log to RAM or die
   } catch (std::bad_alloc const&) {
     if (leading) {
       LOG_TOPIC(FATAL, Logger::AGENCY)
           << "RAFT leader fails to allocate volatile log entries!";
       FATAL_ERROR_EXIT();
     } else {
       LOG_TOPIC(ERR, Logger::AGENCY)
-          << "RAFT follower fails to allocate volatile log entries!";
-      return 0;
+        << "RAFT follower fails to allocate volatile log entries!";
     }
   }
 
-  if (leading) {
+  if (!l.clientId.empty()) {
     try {
       _clientIdLookupTable.emplace(  // keep track of client or die
-          std::pair<std::string, index_t>(clientId, idx));
+        std::pair<std::string, uint64_t>{l.clientId, l.index});
```

Comment:
Could you not use `_clientIdLookupTable.emplace(l.clientId, l.index)` here and avoid a copy or move?

Reply:
I know it has been like this before.

Reply:
of course. thx. but  index_t is of course `uint64_t`
`typedef uint64_t index_t;`